src/seedu/addressbook/data/person/UniquePersonList.java

```diff
@@ -81,6 +82,23 @@ public UniquePersonList(UniquePersonList source) {
         return Collections.unmodifiableList(internalList);
     }
 
+    /**
+     * Returns an unmodifiable java List view sorted in alphabetical order
+     * with elements cast as immutable {@link ReadOnlyPerson}
+     */
+    public List<ReadOnlyPerson> sortImmutableListView() {
+        final List<ReadOnlyPerson> sortedList = new ArrayList<>();
+        sortedList.addAll(internalList);
+        sortedList.sort(new Comparator<ReadOnlyPerson>() {
+            @Override
+            public int compare(ReadOnlyPerson o1, ReadOnlyPerson o2) {
```

Comment:
Maybe we can abstract the compare function outside so that we can use the same Sort function to sort the list in different ways